### ISSUE TITLE
F17 removal lost important tests

### ISSUE THREAD

**Issue by jribbens** [w3c/wcag#1492]
F17 was removed because F77 overlapped with it. However F17 included 4 tests, and only 1 of these overlapped with F77, so its removal lost 3 important tests that still need to be covered by WCAG (non-unique accesskeys, and various types of references to IDs where the ID does not exist). F17 should be reinstated (perhaps with its test 1 removed).

**Comment by JAWS-test** [comment]
In my opinion, duplicate access keys are not prohibited (neither the HTML specification prohibits them nor WCAG SC 4.1.1). The handling of duplicate accesskeys differs from browser to browser, but I would not see a problem from an accessibility point of view

**Comment by patrickhlauke** [comment]
x-ref https://github.com/w3c/wcag/issues/770

**Comment by jribbens** [comment]
> In my opinion, duplicate access keys are not prohibited (neither the HTML specification prohibits them nor WCAG SC 4.1.1). The handling of duplicate accesskeys differs from browser to browser, but I would not see a problem from an accessibility point of view

How could it possibly *not* be an accessibility issue? There is no definition of what should happen in this instance, there is no obviously correct behaviour, there are several potential behaviours (click the first element with the access key, click the last one, click all of them!?), browsers and accessibility tools are likely to behave in differing ways. Duplicate access keys is almost certain to be a mistake by the page author that they need to fix in order for the page feature(s) to be accessible as they intended.

**Comment by patrickhlauke** [comment]
Noting that the normative wording for 4.1.1 does not cover duplicate/conflicting accesskey values

> In content implemented using markup languages, elements have complete start and end tags, elements are nested according to their specifications, elements do not contain duplicate attributes, and any IDs are unique, except where the specifications allow these features.

while yes, duplicate `accesskey` definitions are invalid markup (per https://html.spec.whatwg.org/multipage/interaction.html#the-accesskey-attribute "none of which are identical to another token"), the SC doesn't strictly touch on these types of validation issues

**Comment by JAWS-test** [comment]
> while yes, duplicate accesskey definitions are invalid markup

Sorry, I did not find this part of the specification when searching for accesskey. I.e. duplicate accesskeys are not correct HTML syntax (even if the [HTML validator](https://validator.w3.org/) does not show any error), but still no violation of the WCAG

**Comment by JAWS-test** [comment]
> How could it possibly not be an accessibility issue? 

As long as the WCAG does not require access keys to be used at all (e.g. to simplify keyboard operation, which I would find useful), in my opinion problems with access keys cannot be accessibility problems

**Comment by jribbens** [comment]
> As long as the WCAG does not require access keys to be used at all (e.g. to simplify keyboard operation, which I would find useful), in my opinion problems with access keys cannot be accessibility problems

Do you mean "problems with access keys cannot be WCAG issues"? Because they obviously *are* accessibility problems, the purpose of the `accesskey` feature is to implement accessibility, so a problem with it is *by definition* an accessibility problem. If SC 4.1.1 isn't deemed appropriate then how about 2.1.1 ("Keyboard")? If the page author has deemed that accesskeys are required for accessibility reasons and they are not functioning properly then that seems like a 2.1.1 failure to me.

**Comment by JAWS-test** [comment]
>  the purpose of the accesskey feature is to implement accessibility

I do not think so. Accesskeys were not invented for accessibility, but for efficient operation. For example, I am a mouse user, but I use accesskeys frequently to reach my target faster. Of course, accesskeys can also be useful for accessibility: for keyboard users.

> Do you mean "problems with access keys cannot be WCAG issues"? 

In my opinion, the WCAG does not dictate that a site must have accesskeys or that the accesskeys must work. However, if keyboard operation is only possible with access keys, this would be a violation of SC 2.1.1 or 2.1.2.

But F17 does not refer to SC 2.1.1 or 2.1.2, but to SC 1.3.1 and 4.1.1. And with these two criteria (SC 1.3.1 and 4.1.1) I would not consider double accesskeys as an error.

**Comment by patrickhlauke** [comment]
> > while yes, duplicate accesskey definitions are invalid markup
> 
> Sorry, I did not find this part of the specification when searching for accesskey. I.e. duplicate accesskeys are not correct HTML syntax (even if the [HTML validator](https://validator.w3.org/) does not show any error)

it's the part that i quoted from https://html.spec.whatwg.org/multipage/interaction.html#the-accesskey-attribute

the w3c validator does not check for this, it seems

**Comment by jribbens** [comment]
> > the purpose of the accesskey feature is to implement accessibility
> 
> I do not think so. Accesskeys were not invented for accessibility

Yes, they were. The HTML 4.01 DTD says so explicitly:

```
  accesskey   %Character;    #IMPLIED  -- accessibility key character --
```

> , but for efficient operation. For example, I am a mouse user, but I use accesskeys frequently to reach my target faster. Of course, accesskeys can also be useful for accessibility: for keyboard users.

A very great deal of the the things that come under the heading of "accessibility" help "efficient operation" and are used by a great many users who do not have any accessibility requirements. A more accessible website is more usable. Accesskeys are not special in this respect.

**Comment by mraccess77** [comment]
I would not consider duplicate accesskeys a problem as most implementations I have found work - you just have to press it again to reach the next element which does not cause a failure of any success criteria.

**Comment by jribbens** [comment]
> > > while yes, duplicate accesskey definitions are invalid markup
> > 
> > Sorry, I did not find this part of the specification when searching for accesskey. I.e. duplicate accesskeys are not correct HTML syntax (even if the [HTML validator](https://validator.w3.org/) does not show any error)
> 
> it's the part that i quoted from https://html.spec.whatwg.org/multipage/interaction.html#the-accesskey-attribute
> 
> the w3c validator does not check for this, it seems

To be fair, the spec is not explicit enough to know what is intended. It says "If specified, the value must be an ordered set of unique space-separated tokens none of which are identical to another token" ... it doesn't say whether by "another token" it means "another token in that value" or "another token in any other accesskey value on any other element". The latter is the only interpretation that makes sense, but the former is the only interpretation supported by the actual wording.

**Comment by jribbens** [comment]
> I would not consider duplicate accesskeys a problem as most implementations I have found work - you just have to press it again to reach the next element which does not cause a failure of any success criteria.

That does not work in Edge or Chrome. It does work in Firefox, but only really because Firefox appears to be completely ignoring the HTML specification (pressing the accesskey is supposed to emulate a click on the element, and the most common usage for accesskeys is on links, so if it is following the spec then the browser will have immediately navigated away from the page and so there is no opportunity to "press it again").

**Comment by JAWS-test** [comment]
@jribbens You have convinced me, but in my opinion there is still no WCAG criterion where I can evaluate the double assignment of access keys as a violation of the WCAG, unless it actually prevents keyboard operation (2.1.1., 2.1.2)

**Comment by awkawk** [comment]
I'll agree that accesskey was created with keyboard users in mind, many of whom have disabilities, but the only way that accesskey would be required for a web page or web app is if that was the only way to access some bit of functionality.

Generally accesskey is an additional layer of support for users, so isn't required to conform with WCAG.

**Comment by jribbens** [comment]
> Generally accesskey is an additional layer of support for users, so isn't required to conform with WCAG.

I agree in principle, but as I said above, if the page author has in their infinite human wisdom decided that accesskeys *are* necessary (or at least useful) on a page, saying "well we can see that the access keys you defined are not working as you intended, but we're not going to tell you about it because maybe you were wrong about them being needed" doesn't seem the best or most useful behaviour.

**Comment by awkawk** [comment]
Failures need to always fail, and this doesn't, so it doesn't make sense to include this as a failure.

**Comment by jribbens** [comment]
> Failures need to always fail, and this doesn't, so it doesn't make sense to include this as a failure.

If you think duplicate access keys should be an `H` technique rather than an `F` then I won't argue otherwise... ;-)

**Comment by mraccess77** [comment]
Ok - my memory was from IE and Firefox so thank you for the Chrome/Edge update - IE acts like Firefox as well and focuses the link.   If the conformance claim includes chrome, Edge, or Safari and it prevents access because there is no other way to reach it with the keyboard then that would be a fail of SC 2.1.1 and a failure could be written for that specific situation.
